### PR TITLE
Filter test updates

### PR DESCRIPTION
**Pull request by finestructure**
This removes the `resolveBinds:` parameter from `renderSQL` and adds some additional test integrating the search filters in the package search query.

### CODE REVIEW

File: Tests/AppTests/SearchFilterTests.swift [hunk 1]
```diff
@@ -126,10 +126,10 @@ class SearchFilterTests: AppTestCase {
...
         let builder = SQLSelectBuilder(on: app.db as! SQLDatabase)
             .where(searchFilters: [filter])
-        let sql = renderSQL(builder, resolveBinds: true)
-        _assertInlineSnapshot(matching: sql, as: .lines, with: """
-        SELECT  WHERE ("stars" > '1')
-        """)
+        _assertInlineSnapshot(matching: renderSQL(builder), as: .lines, with: """
+            SELECT  WHERE ("stars" > $1)
+            """)
+        XCTAssertEqual(binds(builder), ["1"])
```
Comment (finestructure) [inline Tests/AppTests/SearchFilterTests.swift:132]: These tests are somewhat redundant now that I added `test_packageMatchQuery_StarsSearchFilter` et al to `SearchTests`. Keeping them in for now but I have some follow-up changes for another day that'll address this.

File: Tests/AppTests/AppTestCase.swift [hunk 2]
```diff
@@ -40,20 +40,14 @@ class AppTestCase: XCTestCase {
 
 
 extension AppTestCase {
-    func renderSQL(_ builder: SQLSelectBuilder?, resolveBinds: Bool = false) -> String {
-        renderSQL(builder?.query, resolveBinds: resolveBinds)
+    func renderSQL(_ builder: SQLSelectBuilder?) -> String {
+        renderSQL(builder?.query)
     }
```
Comment (finestructure) [inline Tests/AppTests/AppTestCase.swift:45]: Removing the `resolveBinds:` parameter forces all tests to assert both the sql with the substitution parameters (`$1`, `$2`, ...) and the binds separately. A bit more work but the big advantage is that we can see and ensure that binds are always used and filled in as we expect.

Comment (daveverwer) [reply Tests/AppTests/AppTestCase.swift:45]: Yea this is a great change. I can see how the original misunderstanding happened!

File: Tests/AppTests/AppTestCase.swift [hunk 3]
```diff
@@ -64,9 +58,12 @@ extension AppTestCase {
...
         return serializer.binds.reduce(into: []) { result, bind in
-            if let bind = bind as? String { result.append(bind) }
-            if let bind = bind as? Date { result.append(DateFormatter.filterParseFormatter.string(from: bind)) }
-            if let bind = bind as? Int { result.append(String(bind)) }
+            switch bind {
+                case let bind as Date:
+                    result.append(DateFormatter.filterParseFormatter.string(from: bind))
+                default:
+                    result.append("\(bind)")
+            }
```
Comment (finestructure) [inline Tests/AppTests/AppTestCase.swift:66]: There was a bug here where a new type (i.e. `Set<PlatformCompatibility>`) would be skipped from the bind variables. This lead me to debugging binds and ending down the rabbit hole with SQLBind and quoting...

Now we just default to `"\(bind)"` as a replacement which will always yield _something_ and we can then specialise it with a new case above.

**Review by daveverwer — APPROVED** [review]

**Pull request merged**